media/src/main/java/edu/artic/media/audio/AudioPlayerService.kt

```diff
@@ -0,0 +1,328 @@
+                    } else {
+                        audioPlayBackStatus.onNext(PlayBackState.Paused(it))
+                    }
+                }
+            }
+        })
+
+        audioControl.subscribe {
+            when (it) {
+                is PlayBackAction.Play -> {
+                    setArticObject(it.audioFile)
+                    player.playWhenReady = true
+                }
+
+                is PlayBackAction.Resume -> {
+                    player.playWhenReady = true
+                }
+
+                is PlayBackAction.Pause -> {
+                    player.playWhenReady = false
+                }
+
+                is PlayBackAction.Stop -> {
+                    val articAudioFile = articObject?.audioCommentary?.first()?.audioFile
+                    articAudioFile?.let {
+                        audioPlayBackStatus.onNext(PlayBackState.Stopped(it))
+                    }
+                    player.stop()
+                }
+
+                is PlayBackAction.Seek -> {
+                    player.seekTo(it.time)
+                }
+            }
+        }.disposedBy(disposeBag)
+    }
+
+    private fun setUpNotificationManager() {
+        playerNotificationManager = PlayerNotificationManager.createWithNotificationChannel(
+                this,
+                Constants.FOREGROUND_CHANNEL_ID,
+                R.string.channel_name,
+                Constants.NOTIFICATION_ID,
+                object : PlayerNotificationManager.MediaDescriptionAdapter {
+                    override fun createCurrentContentIntent(player: Player?): PendingIntent? {
+                        //TODO make it dynamic so that activity that started the audio stream will be the destination of Intent
+                        val notificationIntent = "edu.artic.audio".asDeepLinkIntent()
+                        return PendingIntent.getActivity(this@AudioPlayerService, 0, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT)
+                    }
+
+                    override fun getCurrentContentText(player: Player?): String? {
+                        return null
+                    }
+
+                    override fun getCurrentContentTitle(player: Player?): String {
+                        return articObject?.title.orEmpty()
+                    }
+
+                    override fun getCurrentLargeIcon(player: Player?, callback: PlayerNotificationManager.BitmapCallback?): Bitmap? {
+                        return null
+                    }
+                })
+
+        playerNotificationManager.setNotificationListener(object : PlayerNotificationManager.NotificationListener {
+            override fun onNotificationCancelled(notificationId: Int) {
+            }
+
+            override fun onNotificationStarted(notificationId: Int, notification: Notification?) {
+                startForeground(notificationId, notification)
+            }
+        })
+
+        playerNotificationManager.setUseNavigationActions(false)
```

Comment:
switch to using `playerNotificationManager.apply { }` here. itll clean up the redundant call to same object.